src/test/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/service/prison/UserServiceTest.kt

```diff
@@ -421,4 +422,14 @@ class UserServiceTest {
       )
     }
   }
+
+  @Test
+  fun `should throw exception when username contains @`() {
+    val username = "SOMEUSER@NAME"
+    doThrow(EntityNotFoundException("Prison username $username not found")).whenever(prisonUserApiService).findUserByUsername(username)
```

Comment:
Whilst this test mimics the error, it would be better as an integration test to ensure that the return from the endpoint fails gracefully - with 404